pkg/controller/core/workload_controller.go

```diff
@@ -548,7 +548,8 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 
 func (r *WorkloadReconciler) reconcileSyncAdmissionChecks(ctx context.Context, wl *kueue.Workload, cq *kueue.ClusterQueue) (bool, error) {
 	log := ctrl.LoggerFrom(ctx)
-	admissionChecks := workload.AdmissionChecksForWorkload(log, wl, admissioncheck.NewAdmissionChecks(cq))
+	numAllFlavors := utilslices.Reduce(cq.Spec.ResourceGroups, func(acc int, rg kueue.ResourceGroup) int { return acc + len(rg.Flavors) }, 0)
```

Comment:
Nice trick, but I think we should have a helper function to better capture the intention, and reuse it from all places